# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1221677822 14400
#      Wed Sep 17 14:57:02 2008 -0400
# Node ID 587eee4120ff2652ebbe303a639e12a169abc341
# Parent  fdc7c91835ab7fdc71e60f6237be0d3dd14f0561
cut trailing singeltons in cat function

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,12 @@
 2008-09-17  David Bateman  <dbateman@free.fr>
 
+	* data.cc (do_cat): Chopping trailing singleton dimensions.
+
 	* data.cc (SINGLE_TYPE_CONCAT, DO_SINGLE_TYPE_CONCAT): New macros
 	(do_cat): Special case single type concatenations for speed.
 	* pt.mat.cc (std::string get_concat_class (const std::string&,
 	const std::string&), void maybe_warn_string_concat (bool, bool)):
 	Remove static declaration.
 	* pt-mat.h (std::string get_concat_class (const std::string&,
 	const std::string&), void maybe_warn_string_concat (bool, bool)):
 	Define extern here.
diff --git a/src/data.cc b/src/data.cc
--- a/src/data.cc
+++ b/src/data.cc
@@ -1848,16 +1848,20 @@ do_cat (const octave_value_list& args, s
 		    error ("%s: indexing error", fname.c_str ());
 		  break;
 		}
 	      else
 		ra_idx (dim) += (dim < dv_tmp.length () ? 
 				 dv_tmp (dim) : 1);
 	    }
 
+	  // Reshape, chopping trailing singleton dimensions
+	  dv.chop_trailing_singletons ();
+	  tmp = tmp.reshape (dv);
+
 	  retval = tmp;
 	}
       else
 	error ("%s: invalid dimension argument", fname.c_str ());
     }
   else
     print_usage ();
  
